### ISSUE TITLE
[Q]Exception: The wandb backend process has shutdown?

### ISSUE THREAD

**Issue by wensimokedoudou** [wandb/wandb#3045]
INFO: Model loaded from data_models/checkpoint_epoch200.pth
INFO: Creating dataset with 260 examples
INFO: setting login settings: {}
wandb: Currently logged in as: wensimokedoudou (use `wandb login --relogin` to force relogin)
wandb: wandb version 0.12.7 is available!  To upgrade, please run:
wandb:  $ pip install wandb --upgrade
wandb: Tracking run with wandb version 0.10.8
wandb: Syncing run stellar-firefly-3
wandb: ⭐️ View project at https://wandb.ai/wensimokedoudou/u-net
wandb: 🚀 View run at https://wandb.ai/wensimokedoudou/u-net/runs/175ajqdz
wandb: Run data is saved locally in wandb/run-20211215_195933-175ajqdz
wandb: Run `wandb off` to turn off syncing.

Traceback (most recent call last):
  File "train.py", line 206, in <module>
    amp=args.amp)
  File "train.py", line 64, in train_net
    amp=amp))
  File "/usr/local/lib/python3.6/dist-packages/wandb/sdk/wandb_config.py", line 87, in update
    self._callback(data=self._as_dict())
  File "/usr/local/lib/python3.6/dist-packages/wandb/sdk/wandb_run.py", line 592, in _config_callback
    self._backend.interface.publish_config(data)
  File "/usr/local/lib/python3.6/dist-packages/wandb/interface/interface.py", line 497, in publish_config
    self._publish_config(cfg)
  File "/usr/local/lib/python3.6/dist-packages/wandb/interface/interface.py", line 501, in _publish_config
    self._publish(rec)
  File "/usr/local/lib/python3.6/dist-packages/wandb/interface/interface.py", line 429, in _publish
    raise Exception("The wandb backend process has shutdown")
Exception: The wandb backend process has shutdown

wandb: Waiting for W&B process to finish, PID 52394
wandb: Program failed with code 1.  Press ctrl-c to abort syncing.
wandb: ERROR Problem finishing run
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/wandb/sdk/wandb_run.py", line 1271, in _atexit_cleanup
    self._on_finish()
  File "/usr/local/lib/python3.6/dist-packages/wandb/sdk/wandb_run.py", line 1415, in _on_finish
    self._backend.interface.publish_exit(self._exit_code)
  File "/usr/local/lib/python3.6/dist-packages/wandb/interface/interface.py", line 572, in publish_exit
    self._publish(rec)
  File "/usr/local/lib/python3.6/dist-packages/wandb/interface/interface.py", line 429, in _publish
    raise Exception("The wandb backend process has shutdown")
Exception: The wandb backend process has shutdown
/usr/lib/python3.6/multiprocessing/semaphore_tracker.py:143: UserWarning: semaphore_tracker: There appear to be 6 leaked semaphores to clean up at shutdown
  len(cache))

**Comment by ramit-wandb** [comment]
Hi @wensimokedoudou,

I noticed that you are running `wandb 0.10.8`. We have made a lot of updates and bug fixes since then and the latest release version is `0.12.7`. Could you try upgrading via `pip install wandb --upgrade`? 

Please let us know if the issue persists after your upgrade.

**Comment by yolkandwhite** [comment]
I'm having same issue with `0.12.7`

**Comment by ramit-wandb** [comment]
Hi @Yoonho-Na, what operating system are you using?

**Comment by nzw0301** [comment]
I'm not sure my issue is related to this, but let me share my issue because the error message was same: `Exception: The wandb backend process has shutdown`.

The error message was something like this:

```bash
$ python iterative.py

.......

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/homebrew/Caskroom/miniconda/base/envs/optuna/lib/python3.9/site-packages/wandb/sdk/wandb_run.py", line 1797, in _atexit_cleanup
    self._backend.cleanup()
  File "/opt/homebrew/Caskroom/miniconda/base/envs/optuna/lib/python3.9/site-packages/wandb/sdk/backend/backend.py", line 245, in cleanup
    self.interface.join()
  File "/opt/homebrew/Caskroom/miniconda/base/envs/optuna/lib/python3.9/site-packages/wandb/sdk/interface/interface_shared.py", line 458, in join
    super().join()
  File "/opt/homebrew/Caskroom/miniconda/base/envs/optuna/lib/python3.9/site-packages/wandb/sdk/interface/interface.py", line 599, in join
    _ = self._communicate_shutdown()
  File "/opt/homebrew/Caskroom/miniconda/base/envs/optuna/lib/python3.9/site-packages/wandb/sdk/interface/interface_shared.py", line 455, in _communicate_shutdown
    _ = self._communicate(record)
  File "/opt/homebrew/Caskroom/miniconda/base/envs/optuna/lib/python3.9/site-packages/wandb/sdk/interface/interface_shared.py", line 213, in _communicate
    return self._communicate_async(rec, local=local).get(timeout=timeout)
  File "/opt/homebrew/Caskroom/miniconda/base/envs/optuna/lib/python3.9/site-packages/wandb/sdk/interface/interface_shared.py", line 218, in _communicate_async
    raise Exception("The wandb backend process has shutdown")
Exception: The wandb backend process has shutdown
```

Additional log files are found at 
- [`debug.log`](https://gist.github.com/nzw0301/c598319a37cc9dfdd86c6d1c4e71c576)
- [`debug-internal.log`](https://gist.github.com/nzw0301/5dc36f9efee48fd37030c7147f88c695)
- my code: https://github.com/nzw0301/wandb-optuna/blob/main/iterative.py

wandb version is the latest stable `0.12.9` and the main libraries versions are as follows:

- optuna: 2.10.0
- torch: 1.10.1
- torchvision: 0.11.2

**Comment by nzw0301** [comment]
Let me share the same error with a different start method: default value, which might be `spawn`.

```bash
$ python iterative.py

OMP: Error #178: Function Can't open SHM2 failed:
OMP: System error #24: Too many open files
Problem at: /Users/nzw/Documents/wandb-optuna/iterative.py 85 objective
Traceback (most recent call last):
  File "/opt/homebrew/Caskroom/miniconda/base/envs/optuna/lib/python3.9/site-packages/wandb/sdk/wandb_init.py", line 931, in init
    run = wi.init()
  File "/opt/homebrew/Caskroom/miniconda/base/envs/optuna/lib/python3.9/site-packages/wandb/sdk/wandb_init.py", line 544, in init
    backend.interface.publish_header()
  File "/opt/homebrew/Caskroom/miniconda/base/envs/optuna/lib/python3.9/site-packages/wandb/sdk/interface/interface.py", line 80, in publish_header
    self._publish_header(header)
  File "/opt/homebrew/Caskroom/miniconda/base/envs/optuna/lib/python3.9/site-packages/wandb/sdk/interface/interface_shared.py", line 247, in _publish_header
    self._publish(rec)
  File "/opt/homebrew/Caskroom/miniconda/base/envs/optuna/lib/python3.9/site-packages/wandb/sdk/interface/interface_queue.py", line 45, in _publish
    raise Exception("The wandb backend process has shutdown")
Exception: The wandb backend process has shutdown
wandb: ERROR Abnormal program exit
```

`debug.log` is as follows:

```bash
2022-01-02 05:24:45,857 INFO    MainThread:42958 [wandb_setup.py:_flush():71] setting env: {}
2022-01-02 05:24:45,857 INFO    MainThread:42958 [wandb_setup.py:_flush():71] setting login settings: {}
2022-01-02 05:24:45,857 INFO    MainThread:42958 [wandb_init.py:_log_setup():371] Logging user logs to /Users/nzw/Documents/wandb-optuna/wandb/run-20220102_052445-2x0fvezm/logs/debug.log
2022-01-02 05:24:45,857 INFO    MainThread:42958 [wandb_init.py:_log_setup():372] Logging internal logs to /Users/nzw/Documents/wandb-optuna/wandb/run-20220102_052445-2x0fvezm/logs/debug-internal.log
2022-01-02 05:24:45,857 INFO    MainThread:42958 [wandb_init.py:init():404] calling init triggers
2022-01-02 05:24:45,858 INFO    MainThread:42958 [wandb_init.py:init():409] wandb.init called with sweep_config: {}
config: {'n_layers': 1, 'n_units_l0': 107, 'dropout_l0': 0.25153634360693383, 'optimizer': 'RMSprop', 'lr': 0.006069848966840492, 'trial.number': 21}
2022-01-02 05:24:45,858 INFO    MainThread:42958 [wandb_init.py:init():460] starting backend
2022-01-02 05:24:45,858 INFO    MainThread:42958 [backend.py:_multiprocessing_setup():99] multiprocessing start_methods=spawn,fork,forkserver, using: spawn
2022-01-02 05:24:45,859 INFO    MainThread:42958 [backend.py:ensure_launched():216] starting backend process...
2022-01-02 05:24:45,870 INFO    MainThread:42958 [backend.py:ensure_launched():221] started backend process with pid: 43147
2022-01-02 05:24:45,871 INFO    MainThread:42958 [wandb_init.py:init():469] backend started and connected
2022-01-02 05:24:45,878 INFO    MainThread:42958 [wandb_init.py:init():533] updated telemetry
2022-01-02 05:24:45,885 ERROR   MainThread:42958 [wandb_init.py:init():958] error
Traceback (most recent call last):
  File "/opt/homebrew/Caskroom/miniconda/base/envs/optuna/lib/python3.9/site-packages/wandb/sdk/wandb_init.py", line 931, in init
    run = wi.init()
  File "/opt/homebrew/Caskroom/miniconda/base/envs/optuna/lib/python3.9/site-packages/wandb/sdk/wandb_init.py", line 544, in init
    backend.interface.publish_header()
  File "/opt/homebrew/Caskroom/miniconda/base/envs/optuna/lib/python3.9/site-packages/wandb/sdk/interface/interface.py", line 80, in publish_header
    self._publish_header(header)
  File "/opt/homebrew/Caskroom/miniconda/base/envs/optuna/lib/python3.9/site-packages/wandb/sdk/interface/interface_shared.py", line 247, in _publish_header
    self._publish(rec)
  File "/opt/homebrew/Caskroom/miniconda/base/envs/optuna/lib/python3.9/site-packages/wandb/sdk/interface/interface_queue.py", line 45, in _publish
    raise Exception("The wandb backend process has shutdown")
Exception: The wandb backend process has shutdown
```

In this case, I could not find `debug-internal.log` file.

**Comment by ramit-wandb** [comment]
Hi @nzw0301,

Thank you for your detailed explanation! Are you working in a Distributed Training setting? If you are, you might want to check out [wandb service](https://github.com/wandb/client/blob/master/docs/dev/wandb-service-user.md). It is a new feature that we are developing that is supposed to make `wandb` much more reliable, especially in a distributed process setting.

Could you please try out wandb service and let me know if this fixes your issue?

Thanks,
Ramit

**Comment by nzw0301** [comment]
Hi @ramit-wandb, thank you for your comment! I'm afraid that I'm not using distributed training in my understanding. I ran the code on the CPU without any distributed option of PyTorch.

Anyway, wandb service looks nice, I'll definitely use it in my future project!

**Comment by ramit-wandb** [comment]
I see, another solution that could work for you is to set your `WANDB_START_METHOD` to `"thread"` (especially if you are working on linux). Linux doesn't seem to play well with the default `"spawn"` mode.

As for `service`, it extends beyond just distributed training and is supposed to make `wandb` more reliable in general, so it could be worth trying.

**Comment by nzw0301** [comment]
In fact, the code in the first comment used `"thread"`, precisely, https://github.com/nzw0301/wandb-optuna/blob/968be5b811d1f0841dbcdf4caea3c2bddd6a9ebf/iterative.py#L19. 

Anyway, I'll try using `service`.

Thank you for your comment!

**Comment by hasansalimkanmaz** [comment]
Have you solved the issue?

**Comment by janpf** [comment]
Having the same issue on `0.12.9` using `WANDB_START_METHOD="thread"`

**Comment by hasansalimkanmaz** [comment]
I realized that this was resulting from a long `tag` that I use. I truncate the tag and it worked seamlessly. I think the error should be more explicit than `wandb backend process has shutdown`.

**Comment by janpf** [comment]
ah that's interesting! how long was the tag that caused issues?

**Comment by hasansalimkanmaz** [comment]
It was more than 64 chars. Now, it is less than 64 and it works.

**Comment by janpf** [comment]
The gradients getting autologged with huggingface are very very close to this limit.
Therefor I disabled them (`WANDB_WATCH="false" WANDB_START_METHOD="thread" bash run.sh`) but I believe this characterlimit is not the issue in this case
But thanks for your reply :)

**Comment by will-maclean** [comment]
Having the same issue. Have tested with `service` but getting the same issue. Anyone know any other possible fixes? This is with:
 - `python=3.8`
 - `wandb=0.12.9`
 - `pytorch=1.10.1`

**Comment by vanpelt** [comment]
Hey @will-maclean are you able to share a minimal code example that causes the error to occur?  We'll need to reproduce on our end to see what's going on.

**Comment by armanhar** [comment]
@will-maclean Hey, I wanted to follow up with you regarding your support request as we have not heard back from you. Let me know if you need further assistance.

**Comment by hogru** [comment]
I am also having this issue; the code was working perfectly fine w.r.t. wandb; in the last couple of weeks I changed some code, optimized the model, all with `wandb mode = disabled`. After changing to `online` again I get this error. I use it with pytorch lightning and the only thing (I think) I changed w.r.t. wandb are the hyperparameters I log (but can't say for sure).

Happy to hear any updates (even negative ones) since I need to publish results for a university project and need to decide whether to switch to Tensorboard (at least for the reports).

**Comment by emanuelevivoli** [comment]
I'm having the same issue with the `mmf` framework:

```bash
Traceback (most recent call last):
  File "/home/evivoli/projects/mmf/mmf/utils/logger.py", line 456, in __del__
    self._wandb.finish()
  File "/home/evivoli/miniconda3/envs/mmf/lib/python3.7/site-packages/wandb/sdk/wandb_run.py", line 3322, in finish
    wandb.run.finish(exit_code=exit_code, quiet=quiet)
  File "/home/evivoli/miniconda3/envs/mmf/lib/python3.7/site-packages/wandb/sdk/wandb_run.py", line 256, in wrapper
    return func(self, *args, **kwargs)
  File "/home/evivoli/miniconda3/envs/mmf/lib/python3.7/site-packages/wandb/sdk/wandb_run.py", line 222, in wrapper
    return func(self, *args, **kwargs)
  File "/home/evivoli/miniconda3/envs/mmf/lib/python3.7/site-packages/wandb/sdk/wandb_run.py", line 1678, in finish
    return self._finish(exit_code, quiet)
  File "/home/evivoli/miniconda3/envs/mmf/lib/python3.7/site-packages/wandb/sdk/wandb_run.py", line 1684, in _finish
    tel.feature.finish = True
  File "/home/evivoli/miniconda3/envs/mmf/lib/python3.7/site-packages/wandb/sdk/lib/telemetry.py", line 41, in __exit__
    self._run._telemetry_callback(self._obj)
  File "/home/evivoli/miniconda3/envs/mmf/lib/python3.7/site-packages/wandb/sdk/wandb_run.py", line 572, in _telemetry_callback
    self._telemetry_flush()
  File "/home/evivoli/miniconda3/envs/mmf/lib/python3.7/site-packages/wandb/sdk/wandb_run.py", line 583, in _telemetry_flush
    self._backend.interface._publish_telemetry(self._telemetry_obj)
  File "/home/evivoli/miniconda3/envs/mmf/lib/python3.7/site-packages/wandb/sdk/interface/interface_shared.py", line 74, in _publish_telemetry
    self._publish(rec)
  File "/home/evivoli/miniconda3/envs/mmf/lib/python3.7/site-packages/wandb/sdk/interface/interface_queue.py", line 49, in _publish
    raise Exception("The wandb backend process has shutdown")
Exception: The wandb backend process has shutdown
```

Any advice?

**Comment by mnbucher** [comment]
I've runned into the same issue when running my code on Google Colab after restarting the kernel. At least for me the problem was due to authentication issues (maybe some failure when caching the auth token as I restarted the kernel, was and somehow was already logged in, but wandb.init() failed). I could solve it by running:

`!wandb login --relogin`

When working locally, the "!" char may be omitted via shell as this is jupyter notebook syntax.
